geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryStrategyDecider.scala

```diff
@@ -49,14 +52,24 @@ object QueryStrategyDecider extends QueryStrategyDecider with MethodProfiling {
    */
   override def chooseStrategies(sft: SimpleFeatureType,
                                 query: Query,
-                                hints: StrategyHints,
+                                stats: GeoMesaStats,
                                 requested: Option[StrategyType],
                                 output: ExplainerOutputType): Seq[Strategy] = {
 
     implicit val timings = new Timing()
 
     // get the various options that we could potentially use
-    val options = new QueryFilterSplitter(sft).getQueryOptions(query.getFilter, output)
+    val options = {
+      val all = new QueryFilterSplitter(sft).getQueryOptions(query.getFilter, output)
+      // don't evaluate z2 index if there is a z3 option, it will always be picked
```

Comment:
yeah, there will only be a z3 and z2 option if there is a spatial and a temporal component - slightly confusing in that those checks are done in the query splitter, and not here. The thinking is that eventually we probably want to evaluate both z2 and z3 options when there is a temporal component, as if you search for e.g. 5 years z3 is going to struggle with all the ranges per week. However, since we don't consider that now, we just filter out the z2's up front so we don't have to calculate estimated costs.